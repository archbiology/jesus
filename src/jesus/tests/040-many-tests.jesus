create text light = "Day"
create text messiah = "Jesus"
create number age = 33
create text priest = "me"
create real average = 2.3
say age
say messiah
say priest
warn age
say "adult" if (( age >= 18 ) ) otherwise "minor"
(1 + 1 )
( 2 + 7 )
( 1 + age)
(age + 4)
((( age + 3 + age )))
( age/ ( 3 + 3) * 4 )
1+2*3 == 7
(-1)
---++age
not age
not 1
not not 23
 - + ( 1.3 + 4 )
( age or 7 )
( 0 or 7 )
( 0 or not 13 )
( 0 and 2 )
( age and 7 )
( age and age )
( age and 0 )
( 0 and age )
(0 versus age)
(1 versus age )
( 1 versus 0 )
create text name = "Jesus"
( age versus name )
( 1 vs 9 )
say name if ( 1 vs 1 ) otherwise "final"
(age*4)
say "adult" if ((age>=18)) otherwise "minor"
( age == 4 )
( age == age )
( age > age )
( age >= age )
( age < age )
( age <= age )
( age <= 1 )
( ( age <= 1 ) )

create number age = 33
say "adult" if ( ( age >= 18 ) ) otherwise "minor"
( 1 + 1 )
( 2 + 7 )
( 1 + age )
( age + 4 )
( age + 3 + age )
( age + age + age )
( age - age )
( age - 1 )
( 1 - age )
-1
-1 + age
-1
( age * 4 )
( age / 2 )
( age / 3 )
( age / 3 + 3 * 4 )
( age / ( 3 + 3 ) * 4 )
1 + 2 * 3 == 7
(  -1 )
- - - + + age
not age
not 1
not not 23
 - + ( 1.3 + 4 )
( age or 7 )
( 0 or 7 )
( 0 or not 13 )
( 0 and 2 )
( age and 7 )
( age and age )
( age and 0 )
( 0 and age )
( 0 versus age )
( 1 versus age )
( 1 versus 0 )
( age versus age )
( 1 versus 1 )
( 1 versus 9 )
( 1 versus 7 )
( 1.0 versus 7 )
( 1.0 versus 7.98 )
create text name = "Jesus"
( age versus name )
( 1 vs 9 )
say name if ( 1 vs 1 ) otherwise "final"
(age*4)
say "adult" if ((age>=18)) otherwise "minor"
create with_no_type = 33
create number count = "wrong value"
create number one = 1
say count
say one
create number zero = 0
say zero
create number PI = 3.14
say PI
create number temperature = -6
say temperature
create positive elders = 24
say elders
create positive sinful = 0
say sinful
create negative debt = -1
say debt
create negative credit = 13
say credit
create negative void = 0
say void
create natural spirits = 7
say spirits
create natural darkness = -13
say darkness
create percentage fallen_angels = 0.333
say fallen_angels
create percentage fallen_wise = -0.333
say fallen_wise
create percentage sheep = 100
create percentage sheep = 101
say sheep
create word empty = ""
say empty
create word savior = "Jesus"
say savior
create word invalid = "Sinful wise"
warn invalid
create phrase confusion = 6
warn confusion
create phrase lie = "God does not see"
warn lie
create text valid = ""
say valid
create text rebellion = 13
warn rebellion
create car hilux = 0
say hilux
create decimal forgive = 70 * 7
say forgive
valid = "Jesus loves YOU"
say valid
create number age = 8
create number Positive > 0
create Positive abc = -9
say abc
create Positive gifts = 9
say gifts
create number Negative < 0
create Negative debit = -6
say debit
debit = 19
create number Oter >= 7
create Oter a = 7
say a
create Oter a = 6
create number Tribe <= 12
create Tribe israel = 12
create Tribe israel = 13
say israel
create text Confirm matches "(Y|y)es"
create Confirm sure = "Yes"
sure = "no"
say sure
create text question = "What is your age? "
create number edad = ask question
18
say edad
create number idade = ask "Qual sua idade? "
seven
fourteen
33
say idade
create number year = ask PI
create number year2 = ask temperature
age = ask "What is your age again? "
fourty
eternal
2025
say age
question = ask question
What is the new question?
say question
