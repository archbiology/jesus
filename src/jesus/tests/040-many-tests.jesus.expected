(Jesus) (Jesus) (Jesus) (Jesus) (Jesus) (Jesus) (int) 33
(Jesus) Jesus
(Jesus) me
(Jesus) (int) 33
(Jesus) adult
(Jesus) (int) 2
(Jesus) (int) 9
(Jesus) (int) 34
(Jesus) (int) 37
(Jesus) (int) 69
(Jesus) (double) 22.000000
(Jesus) (logic) 1
(Jesus) (double) -1.000000
(Jesus) (double) -33.000000
(Jesus) (logic) 0
(Jesus) (logic) 0
(Jesus) (logic) 1
(Jesus) (double) -5.300000
(Jesus) (int) 33
(Jesus) (int) 7
(Jesus) (logic) 0
(Jesus) (int) 0
(Jesus) (int) 7
(Jesus) (int) 33
(Jesus) (int) 0
(Jesus) (int) 0
(Jesus) (int) 33
(Jesus) (int) 32
(Jesus) (int) 1
(Jesus) (Jesus) null
(Jesus) (int) 8
(Jesus) final
(Jesus) (int) 132
(Jesus) adult
(Jesus) (logic) 0
(Jesus) (logic) 1
(Jesus) (logic) 0
(Jesus) (logic) 1
(Jesus) (logic) 0
(Jesus) (logic) 1
(Jesus) (logic) 0
(Jesus) (logic) 0
(Jesus) ❌ Error: Unknown expression type (parser.cpp)
(Jesus) ❌ Error: The variable 'age' already exists.
(Jesus) adult
(Jesus) (int) 2
(Jesus) (int) 9
(Jesus) (int) 34
(Jesus) (int) 37
(Jesus) (int) 69
(Jesus) (int) 99
(Jesus) (int) 0
(Jesus) (int) 32
(Jesus) (int) -32
(Jesus) (double) -1.000000
(Jesus) (double) 32.000000
(Jesus) (double) -1.000000
(Jesus) (int) 132
(Jesus) (double) 16.500000
(Jesus) (double) 11.000000
(Jesus) (double) 23.000000
(Jesus) (double) 22.000000
(Jesus) (logic) 1
(Jesus) (double) -1.000000
(Jesus) (double) -33.000000
(Jesus) (logic) 0
(Jesus) (logic) 0
(Jesus) (logic) 1
(Jesus) (double) -5.300000
(Jesus) (int) 33
(Jesus) (int) 7
(Jesus) (logic) 0
(Jesus) (int) 0
(Jesus) (int) 7
(Jesus) (int) 33
(Jesus) (int) 0
(Jesus) (int) 0
(Jesus) (int) 33
(Jesus) (int) 32
(Jesus) (int) 1
(Jesus) (int) 0
(Jesus) (int) 0
(Jesus) (int) 8
(Jesus) (int) 6
(Jesus) (int) 6
(Jesus) (int) 6
(Jesus) ❌ Error: The variable 'name' already exists.
(Jesus) null
(Jesus) (int) 8
(Jesus) final
(Jesus) (int) 132
(Jesus) adult
(Jesus) ❌ Error: Expected new type name after 'create with_no_type'
(Jesus) ❌ Error: Error: Invalid value value "wrong value" to variable 'count' declared as type number
(Jesus) (Jesus) ❌ Error: Undefined variable: count
(Jesus) (int) 1
(Jesus) (Jesus) (int) 0
(Jesus) ❌ Error: Error: Invalid value value "(double) 3.140000" to variable 'PI' declared as type number
(Jesus) ❌ Error: Undefined variable: PI
(Jesus) ❌ Error: Error: Invalid value value "(double) -6.000000" to variable 'temperature' declared as type number
(Jesus) ❌ Error: Undefined variable: temperature
(Jesus) ❌ Error: Unknown base type: 'positive'
(Jesus) ❌ Error: Undefined variable: elders
(Jesus) ❌ Error: Unknown base type: 'positive'
(Jesus) ❌ Error: Undefined variable: sinful
(Jesus) ❌ Error: Unknown base type: 'negative'
(Jesus) ❌ Error: Undefined variable: debt
(Jesus) ❌ Error: Unknown base type: 'negative'
(Jesus) ❌ Error: Undefined variable: credit
(Jesus) ❌ Error: Unknown base type: 'negative'
(Jesus) ❌ Error: Undefined variable: void
(Jesus) (Jesus) (int) 7
(Jesus) ❌ Error: Error: Invalid value value "(double) -13.000000" to variable 'darkness' declared as type natural
(Jesus) ❌ Error: Undefined variable: darkness
(Jesus) (Jesus) (double) 0.333000
(Jesus) ❌ Error: Error: Invalid value value "(double) -0.333000" to variable 'fallen_wise' declared as type percentage
(Jesus) ❌ Error: Undefined variable: fallen_wise
(Jesus) (Jesus) ❌ Error: Error: Invalid value value "(int) 101" to variable 'sheep' declared as type percentage
(Jesus) (int) 100
(Jesus) (Jesus) 
(Jesus) (Jesus) Jesus
(Jesus) (Jesus) Sinful wise
(Jesus) ❌ Error: Error: Invalid value value "(int) 6" to variable 'confusion' declared as type phrase
(Jesus) ❌ Error: Undefined variable: confusion
(Jesus) (Jesus) God does not see
(Jesus) (Jesus) 
(Jesus) ❌ Error: Error: Invalid value value "(int) 13" to variable 'rebellion' declared as type text
(Jesus) ❌ Error: Undefined variable: rebellion
(Jesus) ❌ Error: Unknown base type: 'car'
(Jesus) ❌ Error: Undefined variable: hilux
(Jesus) (Jesus) (int) 490
(Jesus) (Jesus) Jesus loves YOU
(Jesus) ❌ Error: The variable 'age' already exists.
(Jesus) (Jesus) ❌ Error: Expected value > (double) 0.000000
(Jesus) ❌ Error: Undefined variable: abc
(Jesus) (Jesus) (int) 9
(Jesus) (Jesus) (Jesus) (double) -6.000000
(Jesus) ❌ Error: Expected value < (double) 0.000000
(Jesus) (Jesus) (Jesus) (int) 7
(Jesus) ❌ Error: Expected value >= (double) 7.000000
(Jesus) (Jesus) (Jesus) ❌ Error: Expected value <= (double) 12.000000
(Jesus) (int) 12
(Jesus) (Jesus) (Jesus) ❌ Error: Value "no" does not match expected pattern.
(Jesus) Yes
(Jesus) (Jesus) What is your age? (Jesus) (double) 18.000000
(Jesus) Qual sua idade? Validation failed: Invalid number: 'seven'
Qual sua idade? Validation failed: Invalid number: 'fourteen'
Qual sua idade? (Jesus) (double) 33.000000
(Jesus) ❌ Error: Variable 'PI' not found. Are you really sure it was declared?
(Jesus) ❌ Error: Variable 'temperature' not found. Are you really sure it was declared?
(Jesus) What is your age again? Validation failed: Invalid number: 'fourty'
What is your age again? Validation failed: Invalid number: 'eternal'
What is your age again? (Jesus) (double) 2025.000000
(Jesus) What is your age? (Jesus) What is the new question?
(Jesus) (Jesus) (Jesus) (int) 1
(Jesus) 